recipes-app/read-rtc-battery/files/src/readrtcbattery.c

```diff
@@ -0,0 +1,54 @@
+/**
+* Copyright (c) Siemens AG, 2019-2023
+*
+* Authors:
+*  Torsten Hahn <torsten.hahn.extern@siemens.com>
+*
+* This file is subject to the terms and conditions of the MIT License.  See
+* COPYING.MIT file in the top-level directory.
+*
+**/
+#include <sys/ioctl.h>
+#include <stdio.h>
+#include <linux/rtc.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+int main()
+{
+    int fd; // Device file path
+    int batteryStatus; // Status of battery 
+    int retval = 0; // Return value of ioctl function (-1 = Error)
+
+    // 1st step: Open Driver
+    fd = open("/dev/rtc", O_RDONLY);
+
+    // 2nd step: Display error in command line if the device
+    // file wasn't found and close application
+    if (fd < 0) {
+        printf("Device file not found\n");
+        return EXIT_FAILURE;
+    }
+
+    // 3rd step: Read out 
+    retval = ioctl(fd, RTC_VL_DATA_INVALID, &batteryStatus);
```

Comment:
This is not correct, please read https://support.industry.siemens.com/forum/de/en/posts/check-rtc-battery-status-of-iot2050/291926 again.